leakcanary-deobfuscation-gradle-plugin/src/main/java/com/squareup/leakcanary/deobfuscation/LeakCanaryLeakDeobfuscationPlugin.kt

```diff
@@ -0,0 +1,129 @@
+      }
+    }
+  }
+
+  private fun findAndroidVariants(project: Project): DomainObjectSet<BaseVariant> {
+    return try {
+      when (val extension = project.extensions.getByType(BaseExtension::class.java)) {
+        is AppExtension -> extension.applicationVariants as DomainObjectSet<BaseVariant>
+        is LibraryExtension -> extension.libraryVariants as DomainObjectSet<BaseVariant>
+        else -> throwNoAndroidPluginException()
+      }
+    } catch (e: Exception) {
+      throwNoAndroidPluginException()
+    }
+  }
+
+  private fun createLeakCanaryExtension(project: Project): LeakCanaryDeobfuscationExtension {
+    return project.extensions.create("leakCanary", LeakCanaryDeobfuscationExtension::class.java)
+  }
+
+  private fun setupTasks(
+    project: Project,
+    variant: BaseVariant
+  ) {
+    val copyObfuscationMappingFileTaskProvider = project.tasks.register(
+        "leakCanaryCopyObfuscationMappingFor${variant.name.capitalize()}",
+        CopyObfuscationMappingFileTask::class.java
+    ) {
+      it.mappingFile = variant.mappingFile
+      it.mergeAssetsDirectory = variant.mergeAssetsProvider.get()
+          .outputDir.get()
+          .asFile
+
+      val mappingGeneratingTaskProvider =
+        findTaskProviderOrNull(
+            project,
+            "transformClassesAndResourcesWithR8For${variant.name.capitalize()}"
+        ) ?: findTaskProviderOrNull(
+            project,
+            "transformClassesAndResourcesWithProguardFor${variant.name.capitalize()}"
+        ) ?: throwMissingMinifiedVariantException(project)
+
+      it.dependsOn(mappingGeneratingTaskProvider)
+    }
+
+    getPackageTaskProvider(variant).configure {
+      it.dependsOn(copyObfuscationMappingFileTaskProvider)
+    }
+  }
+
+  private fun findTaskProviderOrNull(
+    project: Project,
+    taskName: String
+  ): TaskProvider<Task>? {
+    return try {
+      project.tasks.named(taskName)
+    } catch (proguardTaskNotFoundException: UnknownTaskException) {
+      null
+    }
+  }
+
+  private fun getPackageTaskProvider(variant: BaseVariant): TaskProvider<out DefaultTask> {
+    return when (variant) {
+      is LibraryVariant -> variant.packageLibraryProvider
+      is ApplicationVariant -> variant.packageApplicationProvider
+      else -> throwNoAndroidPluginException()
+    }
+  }
+
+  private fun throwNoAndroidPluginException(): Nothing {
+    throw GradleException(
+        "LeakCanary deobfuscation plugin can be used only in Android application or library module."
+    )
+  }
+
+  private fun throwMissingMinifiedVariantException(project: Project): Nothing {
+    project.logger.log(
+        DEBUG,
+        """
+          None of the project's variants seem to have minification enabled. 
+          Please make sure that there is at least 1 minified variant in your project.    
+        """.trimIndent()
+    )
+    throw GradleException(
+        "LeakCanary deobfuscation plugin couldn't find any variant with minification enabled."
```

Comment:
Why not provide all the context directly in the error message? Developers might not know to turn on DEBUG